src/Swift.Runtime/src/Swift/SwiftOptional.cs

```diff
@@ -43,7 +43,7 @@ static TypeMetadata ISwiftObject.GetTypeMetadata()
     /// <summary>
     /// Creates a new SwiftOptional from a Swift payload
     /// </summary>
-    static ISwiftObject ISwiftObject.NewFromPayload(IntPtr payload)
+    static ISwiftObject ISwiftObject.NewFromPayload(SwiftHandle payload)
```

Comment:
Is there a functional difference when using SwiftHandle instead of IntPtr, or not yet? I am trying to understand the benefits of the SwiftHandle abstraction.

Reply:
The difference boils down mainly to handling conflicts in generated code. Lets assume we have a struct `Foo` which on swift side has a single `Int` field. We represent `Int` as `IntPtr` on C# side. 

```csharp
struct Foo
{
    Foo(IntPtr) {} // Constructor accepting single integer of sized determined by target platform
    Foo(SwiftHandle handle) {} // Constructor accepting a pointer
}
```